torch/csrc/jit/unpickler.cpp

```diff
@@ -456,6 +457,16 @@ void Unpickler::readGlobal(
                 " has no tensor table\n");
             stack_.emplace_back(tensor_table_->at(data.toInt()));
       });
+    } else if (class_name == "restore_type_tag") {
+      globals_.emplace_back([this] {
+        auto data = stack_.back().toTuple()->elements();
+        stack_.pop_back();
+        TypePtr type = c10::parseType(data.at(1).toStringRef());
+        // TODO: Use lookahead to avoid creating the tuple and immediately
+        // destroying it here
+        restoreAccurateTypeTags(data.at(0), type);
```

Comment:
This function is recursive and expensive so instead we should just directly restore the tag on the list or dict. The restoreAccurateTypeTags will only be used for older models.